src/java.desktop/share/classes/javax/swing/plaf/nimbus/skin.laf

```diff
@@ -27248,7 +27248,7 @@
          <borderStates/>
          <regions/>
       </uiComponent>
-      <uiComponent opaque="false" type="javax.swing.JToolTip" name="ToolTip" ui="ToolTipUI" subregion="false">
+      <uiComponent opaque="true" type="javax.swing.JToolTip" name="ToolTip" ui="ToolTipUI" subregion="false">
```

Comment:
In my opinion, instead of changing skin.laf we probably should update SynthToolTipUI.installDefaults() and add 
`LookAndFeel.installProperty(c, "opaque", Boolean.TRUE);`

similar to what we do for BasicToolTipUI.installDefaults. Will it not work?

Reply:
No, it doesn't work, i have checked.
The change needs to be skin.laf file only.

Reply:
I am not sure where you placed the code. I think it's need to be added after updateStyle() as updateStyle() uninstall the default and install for new style.
```
protected void installDefaults(JComponent c) {
        updateStyle(c);
    }
```

Reply:
yeah setting the property after updateStyle(c) works fine.
updating the change in the SynthToolTipUI.installDefaults()

Reply:
What is the difference from setting it via "skin.laf"? The "skin.laf" is converted to the NimbusDefaults during the build and these defaults are used by the NimbusLookAndFeel.

Reply:
I personally feel setting in code will be easier to maintain going forward than to go through property file settings.

Reply:
Then why we need this file at all? Now we will set the false value in the config, and true at runtime, what is the profit?

Reply:
I think we should set it in the config file since this is the place where all other properties are stored.

Reply:
Before this fix, I see it was set in property file and then overridden in JToolTip. Current fix does the same, set in property file and then overridden in SynthToolTipUI. But I don't have objection if it is only controlled by property file.